ssl/statem/statem_srvr.c

```diff
@@ -437,6 +437,11 @@ static WRITE_TRAN ossl_statem_server13_write_transition(SSL *s)
             st->hand_state = TLS_ST_SW_CERT_REQ;
             return WRITE_TRAN_CONTINUE;
         }
+        if (s->ext.extra_tickets_expected > 0) {
+            st->hand_state = TLS_ST_SW_SESSION_TICKET;
+            s->ext.extra_tickets_expected--;
```

Comment:
Is it feasible to move this decrement into the ticket processing itself?  I've tried (not 100% successfully) to avoid side effects in the transition functions as much as possible.

Reply:
Sure; I had gone this route since I didn't want to put the "random side effect" into the ticket-generation function, but avoiding side-effects in the transition functions seems like a better thing to prioritize.